unittest/src/fake/buttons/mod.rs

```diff
@@ -0,0 +1,118 @@
+}
+
+impl<const NUM_BUTTONS: usize> Buttons<NUM_BUTTONS> {
+    pub fn new() -> std::rc::Rc<Buttons<NUM_BUTTONS>> {
+        #[allow(clippy::declare_interior_mutable_const)]
+        const OFF: Cell<ButtonState> = Cell::new(ButtonState {
+            pressed: false,
+            interrupt_enabled: false,
+        });
+        std::rc::Rc::new(Buttons {
+            buttons: [OFF; NUM_BUTTONS],
+        })
+    }
+}
+
+impl<const NUM_BUTTONS: usize> crate::fake::SyscallDriver for Buttons<NUM_BUTTONS> {
+    fn id(&self) -> u32 {
+        DRIVER_NUMBER
+    }
+    fn num_upcalls(&self) -> u32 {
+        1
+    }
+
+    fn command(&self, command_number: u32, argument0: u32, _argument1: u32) -> CommandReturn {
+        match command_number {
+            BUTTONS_COUNT => crate::command_return::success_u32(NUM_BUTTONS as u32),
+            BUTTONS_ENABLE_INTERRUPTS => {
+                if argument0 < NUM_BUTTONS as u32 {
+                    let button = self.buttons[argument0 as usize].get();
+                    self.buttons[argument0 as usize].set(ButtonState {
+                        interrupt_enabled: true,
+                        ..button
+                    });
+                    crate::command_return::success()
+                } else {
+                    crate::command_return::failure(ErrorCode::Invalid)
+                }
+            }
+            BUTTONS_DISABLE_INTERRUPTS => {
+                if argument0 < NUM_BUTTONS as u32 {
+                    let button = self.buttons[argument0 as usize].get();
+                    self.buttons[argument0 as usize].set(ButtonState {
+                        interrupt_enabled: false,
+                        ..button
+                    });
+                    crate::command_return::success()
+                } else {
+                    crate::command_return::failure(ErrorCode::Invalid)
+                }
+            }
+            BUTTONS_READ => {
+                if argument0 < NUM_BUTTONS as u32 {
+                    crate::command_return::success_u32(
+                        self.buttons[argument0 as usize].get().pressed as u32,
+                    )
+                } else {
+                    crate::command_return::failure(ErrorCode::Invalid)
+                }
+            }
+            _ => crate::command_return::failure(ErrorCode::NoSupport),
+        }
+    }
+}
+
+// -----------------------------------------------------------------------------
+// Implementation details below
+// -----------------------------------------------------------------------------
+
+#[cfg(test)]
+mod tests;
+
+const DRIVER_NUMBER: u32 = 3;
+
+// Command IDs
+const BUTTONS_COUNT: u32 = 0;
+
+const BUTTONS_ENABLE_INTERRUPTS: u32 = 1;
+const BUTTONS_DISABLE_INTERRUPTS: u32 = 2;
+
+const BUTTONS_READ: u32 = 3;
+
+impl<const NUM_BUTTONS: usize> Buttons<NUM_BUTTONS> {
+    pub fn set_pressed(&self, button: u32, pressed: bool) -> Result<(), ErrorCode> {
```

Comment:
`set_pressed` is a public API, not an implementation detail, correct?

Reply:
Yes.

Reply:
Then it probably belongs above

```
// -----------------------------------------------------------------------------
// Implementation details below
// -----------------------------------------------------------------------------
```

Reply:
Looking at your other PRs, I think you and I understand:

```rust
// -----------------------------------------------------------------------------
// Implementation details below
// -----------------------------------------------------------------------------
```

very differently. What I intend with that comment is to separate the public API of the module from non-public parts of the module.

Perhaps we should change the text of the comment to be clearer? Something like:

```
// -----------------------------------------------------------------------------
// Public API above, module-internal details below.
// -----------------------------------------------------------------------------
```